Controls/Chart/Chart.UWP/Visualization/Common/ContainerVisualsFactory.cs

```diff
@@ -0,0 +1,481 @@
+        {
+            if (orientation == Orientation.Horizontal)
+            {
+                containerVisual.Offset = new Vector3((float)layoutSlot.X, (float)layoutSlot.Bottom, 0);
+                if (containerVisual.Size.X != layoutSlot.Width || containerVisual.Size.Y != 1)
+                {
+                    containerVisual.Size = new Vector2((float)layoutSlot.Width, 1);
+
+                    if (this.dashArrayCache != null)
+                    {
+                        var oldDashesSize = containerVisual.Children.Count * (dashArrayCache[0] + dashArrayCache[1]);
+                        if (containerVisual.Children.Count > 0 && oldDashesSize < layoutSlot.Width)
+                        {
+                            this.ArrangeChartGridDashes(containerVisual, orientation, dashArray, (int)oldDashesSize);
+                        }
+                    }
+                }
+            }
+            else
+            {
+                containerVisual.Offset = new Vector3((float)layoutSlot.X, (float)layoutSlot.Y, 0);
+                if (containerVisual.Size.X != 1 || containerVisual.Size.Y != layoutSlot.Height)
+                {
+                    containerVisual.Size = new Vector2(1, (float)layoutSlot.Height);
+
+                    if (this.dashArrayCache != null)
+                    {
+                        var oldDashesSize = containerVisual.Children.Count * (dashArrayCache[0] + dashArrayCache[1]);
+                        if (containerVisual.Children.Count > 0 && oldDashesSize < layoutSlot.Height)
+                        {
+                            this.ArrangeChartGridDashes(containerVisual, orientation, dashArray, (int)oldDashesSize);
+                        }
+                    }
+                }
+            }
+
+            if (dashArray[0] > 0 && dashArray[1] > 0)
+            {
+                if (containerVisual.Children.Count == 0)
+                {
+                    this.ArrangeChartGridDashes(containerVisual, orientation, dashArray);
+                    this.dashArrayCache = dashArray;
+                }
+                else if (dashArrayCache != null && (dashArrayCache[0] != dashArray[0] || dashArrayCache[1] != dashArray[1]))
+                {
+                    containerVisual.Children.RemoveAll();
+                    this.ArrangeChartGridDashes(containerVisual, orientation, dashArray);
+                    this.dashArrayCache = dashArray;
+                }
+            }
+
+            if (dashArray[0] > 0 && dashArray[1] == 0)
+            {
+                if (containerVisual.Children.Count > 0)
+                {
+                    containerVisual.Children.RemoveAll();
+                }
+
+                this.SetCompositionColorBrush(containerVisual, new SolidColorBrush(Color.FromArgb(0x23, 0, 0, 0)));
```

Comment:
I was not able to find an easy way how to get the exact same resource used by the visual elements. Also, for some values a named brushes are used while others are directly set to the Setters of the Styles to the Background/Fill/Foreground properties. 

Constants according to me are more appropriate - there are only 2-3 brushes that are used.